core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2Endpoints.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.server.security.oauth2;
+
+import javax.inject.Inject;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class OAuth2Endpoints
+{
+    private final Optional<String> accessTokenIssuer;
+    private final String authUrl;
+    private final String tokenUrl;
+    private final String jwksUrl;
+    private final Optional<String> userinfoUrl;
+
+    @Inject
```

Comment:
Please remove this injection. It one path it is used and in other it is not used. I think it would be more clear when this bean would be provided explicitly in both cases.